File: Offer/2249/main.py
from typing import List


class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if root == None:
            return []
        
        res: List[List[int]] = []
        tempArray: List[int]
        queue: List[TreeNode] = [root]
        tempQueue: List[TreeNode]
        node: TreeNode
        i, length = 0, 0
        isLToR = True

        while len(queue) != 0:
            tempArray = []
            tempQueue = []
            length = len(queue)
            if isLToR:
                for i in range(length):
                    node = queue[i]
                    tempArray.append(node.val)
                    if node.left != None:
                        tempQueue.append(node.left)
                    if node.right != None:
                        tempQueue.append(node.right)
            else:
                for i in range(length):
                    tempArray.append(queue[length - 1 - i].val)
                    node = queue[i]
                    if node.left != None:
                        tempQueue.append(node.left)
                    if node.right != None:
                        tempQueue.append(node.right)
            res.append(tempArray)
            queue = tempQueue
            isLToR = not isLToR
        return res
        

File: Offer/2249/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	if root == nil {
		return [][]int{}
	}

	var res [][]int
	var tempArray []int
	var queue = []*TreeNode{root}
	var tempQueue []*TreeNode
	var node *TreeNode

	var isLToR = true
	var i, length int

	for len(queue) != 0 {
		tempArray = []int{}
		tempQueue = []*TreeNode{}
		length = len(queue) - 1
		if isLToR {
			for i = 0; i < length; i++ {
				node = queue[i]
				tempArray = append(tempArray, node.Val)
				if node.Left != nil {
					tempQueue = append(tempQueue, node.Left)
				}
				if node.Right != nil {
					tempQueue = append(tempQueue, node.Right)
				}
			}
		} else {
			for i = 0; i < length; i++ {
				tempArray = append(tempArray, queue[length-1-i].Val)
				node = queue[i]
				if node.Left != nil {
					tempQueue = append(tempQueue, node.Left)
				}
				if node.Right != nil {
					tempQueue = append(tempQueue, node.Right)
				}
			}
		}
		res = append(res, tempArray)
		queue = tempQueue
		isLToR = !isLToR
	}
	return res
}
